cmake/config.cmake.in

```diff
@@ -0,0 +1,28 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+@PACKAGE_INIT@
+
+cmake_minimum_required(VERSION @CMAKE_MINIMUM_REQUIRED_VERSION@)
```

Comment:
That's probably in the fetch content version picking up 3.25 from the example cml due to 
> The <min> version of CMake given to the most recent call to the [cmake_minimum_required(VERSION)](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html#command:cmake_minimum_required) command in the current variable scope or any parent variable scope.

Reply:
Yeah no specific reason. I'm usually running a very recent version of CMake so I set this, but we can definitely use earlier. I bumped up from 3.11 to make my job easier is all.

It looks like the examples CI just uses the default on the ubuntu-latest. Do we have any testing that validates that we can run on a specific version of CMake? Ideally CI would be configured to run on the oldest supported version of CMake so that we have confirmation that we're not using any newer features.

Also, I'll note that when we get around to similar improvements for the device extension we may want to require a newer CMake. There have been a ton of improvements to CMake/CUDA integration. At minimum I would say we should require 3.17 so that we can use FindCUDAToolkit instead of FindCUDA.

Reply:
Everything here runs with 3.12, so I'll just set it to that (older than that requires setting policies, which would clutter the example).